components/kubeflow-pipeline/pipeline_components/analytics-server/flwr_analytics_server/correlation.py

```diff
@@ -30,11 +31,13 @@ def client_input_data(self) -> Dict[str, Scalar]:
     def add_client_data(self, properties: Properties) -> None:
         self._client_data.append(
             AggregationData(
-                num_features=properties["features"],
-                num_entries=properties["entries"],
-                sums=scalar_to_numpy(properties["sums"]),
-                multiply_sums=scalar_to_numpy(properties["multiply_sums"]),
-                variances=scalar_to_numpy(properties["variances"]),
+                num_features=cast(int, properties["features"]),
```

Comment:
Instead of these `cast`ings why don't we actually use an `int`, then we will be sure that the type is correct on runtime and "compile" time:

```suggestion
                num_features=int(properties["features"]),
```

This applies to the other instances where this is seen.